pkg/cdi/cache.go

```diff
@@ -254,6 +259,19 @@ func (c *Cache) InjectDevices(ociSpec *oci.Spec, devices ...string) ([]string, e
 	return nil, nil
 }
 
+func (c *Cache) UpdateMounts(ociSpec *oci.Spec, mounts ...string) ([]string, error) {
```

Comment:
This function is incorrectly named. It does not only update the mounts. It performs the same operations as `InjectDevices` with the exception that unresolved *devices* are output instead of returning an error.

In theory, I would not be against being able to control whether allowing unresolved devices when performing injection, but this is not the way I would propose to do it. Exposing an option at construction time and allowing a caller to retrieve the list of unresolved devices could be a possible solution.

Reply:
Thx @elezar I have updated my code. Please review it again